### ISSUE TITLE
Implement snapshot and garbage collector that move all confirmed blocks to disk, and moves all spent outputs (and inputs?) to disk.

### ISSUE THREAD

**Issue by kulpreet** [kulpreet/libbitcoin-mvcc-database#13]

**Comment by kulpreet** [comment]
This will require a revisit to block, tuple and varlens design. Specifically to handle pointer swizzling when these are pushed to and from disk.

Data is swizzled out to disk only once. 

According to current design, when we need to read transactions, we will populate a temporary buffer to respond to query with data fetched from disk and then blow that away when memory is needed to respond to next query. 

In other words, we don't populate the results fetched from disk into memory block pool, which will require us to once again swizzle and send data to disk. Instead, once data is gone to disk, it never hits the block pool buffers again. Results are constructed in a separate memory space which can just be cleared out whenever we need space for a new query.
